### PR TITLE
Add empty source

### PR DESCRIPTION
**Pull request by fmassot**
Add empty source. Could be useful for testing mainly or if you want to launch a long-running job.

### CODE REVIEW

File: quickwit-indexing/src/source/empty_source.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use async_trait::async_trait;
+use quickwit_actors::{ActorExitStatus, Mailbox, HEARTBEAT};
+use serde::{Deserialize, Serialize};
+
+use crate::models::IndexerMessage;
+use crate::source::{Source, SourceContext, TypedSourceFactory};
+
+pub struct EmptySource;
```
Comment (guilload) [inline quickwit-indexing/src/source/empty_source.rs:27]: `VoidSource` sounds cooler, right? :D

File: quickwit-indexing/src/source/empty_source.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct FilePosition {
```
Comment (guilload) [inline quickwit-indexing/src/source/empty_source.rs:30]: This does not seem to be used.

File: quickwit-indexing/src/source/empty_source.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        tokio::time::sleep(HEARTBEAT / 2).await;
+        Ok(())
+    }
+
+    fn name(&self) -> String {
+        "EmptySource".to_string()
+    }
+
+    fn observable_state(&self) -> serde_json::Value {
+        serde_json::to_value(0).unwrap()
```
Comment (guilload) [inline quickwit-indexing/src/source/empty_source.rs:50]: nit: an empty object `{}` seems more natural IMO.

**Review by guilload — APPROVED** [review]

File: quickwit-indexing/src/source/empty_source.rs (hunk 1 above)
Comment (fmassot) [reply quickwit-indexing/src/source/empty_source.rs:27]: indeed!!!!

File: quickwit-indexing/src/source/empty_source.rs (hunk 3 above)
Comment (fulmicoton) [reply quickwit-indexing/src/source/empty_source.rs:50]:
```suggestion
        serde_json::Value::Object(Default::default())
```

File: quickwit-indexing/src/source/empty_source.rs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EmptySourceParams {}
```
Comment (fulmicoton) [inline quickwit-indexing/src/source/empty_source.rs:55]:
```suggestion
pub struct EmptySourceParams;
```

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**